ocaml/xapi/xapi_guest_agent.ml

```diff
@@ -254,8 +308,9 @@ let all (lookup: string -> string option) (list: string -> string list) ~__conte
     end;
 
 	  (* Update the 'up to date' flag afterwards *)
-	  let gmr = Db.VM_guest_metrics.get_record_internal ~__context ~self:gm in
-	  let up_to_date = Xapi_pv_driver_version.is_up_to_date (Xapi_pv_driver_version.of_guest_metrics (Some gmr)) in
+	  (* Only when both networks and disks paths are optimized, the 'up to date' flag is set to true. *)
+	  let up_to_date = Db.VM_guest_metrics.get_storage_paths_optimized ~__context ~self:gm &&
+	                   Db.VM_guest_metrics.get_network_paths_optimized ~__context ~self:gm in
 	  Db.VM_guest_metrics.set_PV_drivers_up_to_date ~__context ~self:gm ~value:up_to_date;
```

Comment:
Can't we just do `let up_to_date = network_paths_optimized && storage_paths_optimized` rather than getting those flags from the DB, since we've just computed them above?

Also, could we add PV_drivers_up_to_date to the cache, and only update it if changed, as done for other metrics above? We don't need a complete guest metrics record anymore to compute this field.